src/MeshTransfer.f90

```diff
@@ -0,0 +1,1570 @@
+        dx=(union(iun)-union(iun-1))
+        IF(iout>0) THEN
+          IF(iin == 0) THEN
+            TM(iout,1)=TM(iout,1)+dx
+          ELSEIF(iin == SIZE(mesh_in)) THEN
+            TM(iout,SIZE(mesh_in))=TM(iout,SIZE(mesh_in))+dx
+          ELSE
+            TM(iout,iin)=TM(iout,iin)+ &
+                HALF*dx*(TWO*mesh_in(iin+1)-union(iun)-union(iun-1))/ &
+                                (mesh_in(iin+1)-mesh_in(iin))
+            TM(iout,iin+1)=TM(iout,iin+1)+ &
+                HALF*dx*(union(iun)+union(iun-1)-TWO*mesh_in(iin))/   &
+                                (mesh_in(iin+1)-mesh_in(iin))
+          ENDIF
+          vsum=vsum+dx
+        ENDIF
+        IF(iin < SIZE(mesh_in)) THEN
+          IF(union(iun) .APPROXEQ. mesh_in(iin+1)) iin=iin+1
+        ENDIF
+        IF(union(iun) .APPROXEQ. mesh_out(iout+1)) THEN
+          IF(iout>0 .AND. vsum>ZERO) TM(iout,:)=TM(iout,:)/vsum
+          vsum=ZERO
+          iout=iout+1
+          IF(iout>=SIZE(mesh_out)) EXIT
+        ENDIF
+      ENDDO
+
+      ENSURE(ALLOCATED(TM))
+    ENDSUBROUTINE setupP2V_cart
+!
+!-------------------------------------------------------------------------------
+!> @brief  setup point to continuous transfer for cartesian mesh (not type bound)
+!> @param TLS     The direct linear solver object to be set up
+!> @param mesh_in The incoming point mesh
+!> @param minX    The minimum x value for polynomial renormalization
+!> @param maxX    The maximum x value for polynomial renormalization
+!>
+!> This routine initializes a direct linear solver using LU decomposition.  This
+!> routine sets up the linear system to solve the full number of moments.  The
+!> transfer routine will drop moments that aren't needed.  This is inconsequential
+!> because the polynomials are orthogonal.
+!>
+    SUBROUTINE setupP2C_cart(TLS,mesh_in,minX,maxX)
+      TYPE(LinearSolverType_Direct),INTENT(INOUT) :: TLS
+      REAL(SRK),INTENT(IN) :: mesh_in(:)
+      REAL(SRK),INTENT(IN) :: minX
+      REAL(SRK),INTENT(IN) :: maxX
+
+      INTEGER(SIK) :: i,j,nin
+      REAL(SRK) :: h,cent
+      REAL(SRK),ALLOCATABLE :: x(:)
+      TYPE(ParamType) :: myPL
+
+      REQUIRE(.NOT. TLS%isInit)
+      REQUIRE(SIZE(mesh_in)>0)
+
+      nin=SIZE(mesh_in)
+      h=maxX-minX
+      cent=0.5_SRK*(maxX+minX)
+
+      CALL myPL%add('LinearSolverType->TPLType',NATIVE)
+      CALL myPL%add('LinearSolverType->solverMethod',LU)
+      CALL myPL%add('LinearSolverType->MPI_Comm_ID',PE_COMM_SELF)
+      CALL myPL%add('LinearSolverType->numberOMP',1_SNK)
+      CALL myPL%add('LinearSolverType->timerName','testTimer')
+      CALL myPL%add('LinearSolverType->matType',DENSESQUARE)
+      CALL myPL%add('LinearSolverType->A->MatrixType->isSym',.FALSE.)
+      CALL myPL%add('LinearSolverType->A->MatrixType->n',nin)
+      CALL myPL%add('LinearSolverType->x->VectorType->n',nin)
+      CALL myPL%add('LinearSolverType->b->VectorType->n',nin)
+      CALL TLS%init(myPL)
+      CALL myPL%clear()
+
+      ALLOCATE(x(nin))
```

Comment:
in quite a few of these, you allocate x.  It might automatically get cleaned up since it's a local real, but might be good to deallocate it/them for completeness

Reply:
Done... lots of them